agent/consul/auto_encrypt.go

```diff
@@ -130,13 +145,17 @@ func resolveAddr(rawHost string, logger *log.Logger) ([]net.IP, error) {
 	if ips, err := tcpLookupIP(host, logger); err != nil {
 		logger.Printf("[DEBUG] agent: TCP-first lookup failed for '%s', falling back to UDP: %s", host, err)
 	} else if len(ips) > 0 {
-		return ips, nil
+		return ips, port, nil
 	}
 
 	// If TCP didn't yield anything then use the normal Go resolver which
 	// will try UDP, then might possibly try TCP again if the UDP response
 	// indicates it was truncated.
-	return net.LookupIP(host)
+	ips, err := net.LookupIP(host)
```

Comment:
Heh not your code but the logic here seems odd - manually try TCP DNS first but then `net.LookupIP` will try it anyway if response is truncated? I guess in case TCP works and there are more IPs than fit in a UDP packet we save a UDP round trip first to discover that but that seems like a micro-optimization 😄.

Ah well. No point changing it in this PR.